rclpy/test/test_parameter_client.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tempfile import NamedTemporaryFile
+import unittest
+
+import rcl_interfaces.msg
+from rcl_interfaces.msg import ParameterType
+import rcl_interfaces.srv
+import rclpy
+import rclpy.context
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.parameter import Parameter
+from rclpy.parameter_client import AsyncParameterClient
+
+
+class TestParameterClient(unittest.TestCase):
+
+    def setUp(self):
+        self.context = rclpy.context.Context()
+        rclpy.init(context=self.context)
+        self.client_node = rclpy.create_node('test_parameter_client',
+                                             namespace='/rclpy',
+                                             context=self.context)
+        self.target_node = rclpy.create_node('test_parameter_client_target',
+                                             namespace='rclpy',
+                                             allow_undeclared_parameters=True,
+                                             context=self.context)
+        self.target_node.declare_parameter('int_arr_param', [1, 2, 3])
+        self.target_node.declare_parameter('float/param//', 3.14)
```

Comment:
I'm not sure what the slashes in the name are trying to test, but parameter namespaces are separated by period characters (`.`). So, it might make more sense to rename this to exercise the client with a parameter containing a prefix:
```suggestion
        self.target_node.declare_parameter('float.param', 3.14)
```